clap_generate/tests/completions/mod.rs

```diff
@@ -23,10 +23,16 @@ macro_rules! assert_eq {
     };
 }
 
-pub fn common<G: Generator>(app: &mut App, name: &str, fixture: &str) {
+pub fn common<G: Generator>(gen: G, app: &mut App, name: &str, fixture: &str) {
     let mut buf = vec![];
-    generate::<G, _>(app, name, &mut buf);
+    generate(gen, app, name, &mut buf);
     let string = String::from_utf8(buf).unwrap();
 
-    assert_eq!(&string, fixture);
+    assert_eq!(&normalize(&string), &normalize(fixture));
+}
+
+fn normalize(string: &str) -> String {
```

Comment:
The reason we don't normalize is:

1. We want to actually work towards removing those trailing whitespaces and having them in tests shows what else is left
2. We know exactly what's generated at the end

Reply:
I can see that but
- This raises the cost for people to contribute to `clap` which is already pretty high with the complexity and number of non-obvious behavior dependencies.
- Its easy to remove normalization when working to get rid of it
- I can't see us getting much value out of knowing what exactly we generate, down to trailing whitespace,